src/main/java/teammates/common/datatransfer/attributes/FeedbackSessionAttributes.java

```diff
@@ -193,109 +193,68 @@ public String getJsonString() {
-        if (!error.isEmpty()) {
-            errors.add(error);
-        }
+        errors.add(validator.getValidityInfoForNonNullField("feedback session name", feedbackSessionName));
 
-        error = validator.getValidityInfoForNonNullField("course ID", courseId);
-        if (!error.isEmpty()) {
-            errors.add(error);
-        }
+        errors.add(validator.getValidityInfoForNonNullField("course ID", courseId));
 
-        error = validator.getValidityInfoForNonNullField("instructions to students", instructions);
-        if (!error.isEmpty()) {
-            errors.add(error);
-        }
+        errors.add(validator.getValidityInfoForNonNullField("instructions to students", instructions));
 
-        error = validator.getValidityInfoForNonNullField("time for the session to become visible", sessionVisibleFromTime);
-        if (!error.isEmpty()) {
-            errors.add(error);
-        }
+        errors.add(validator.getValidityInfoForNonNullField(
+                "time for the session to become visible", sessionVisibleFromTime));
 
-        error = validator.getValidityInfoForNonNullField("creator's email", creatorEmail);
-        if (!error.isEmpty()) {
-            errors.add(error);
-        }
+        errors.add(validator.getValidityInfoForNonNullField("creator's email", creatorEmail));
 
-        error = validator.getValidityInfoForNonNullField("session creation time", createdTime);
-        if (!error.isEmpty()) {
-            errors.add(error);
-        }
+        errors.add(validator.getValidityInfoForNonNullField("session creation time", createdTime));
 
         // Early return if any null fields
+        errors = getNonEmptyErrors(errors);
         if (!errors.isEmpty()) {
             return errors;
         }
 
-        error = validator.getInvalidityInfoForFeedbackSessionName(feedbackSessionName);
-        if (!error.isEmpty()) {
-            errors.add(error);
-        }
+        errors.add(validator.getInvalidityInfoForFeedbackSessionName(feedbackSessionName));
 
-        error = validator.getInvalidityInfoForCourseId(courseId);
-        if (!error.isEmpty()) {
-            errors.add(error);
-        }
+        errors.add(validator.getInvalidityInfoForCourseId(courseId));
 
-        error = validator.getInvalidityInfoForEmail(creatorEmail);
-        if (!error.isEmpty()) {
-            errors.add("Invalid creator's email: " + error);
-        }
+        errors.add(validator.getInvalidityInfoForEmail(creatorEmail));
 
         // Skip time frame checks if session type is private.
+        errors = getNonEmptyErrors(errors);
         if (this.isPrivateSession()) {
             return errors;
         }
 
-        error = validator.getValidityInfoForNonNullField("submission opening time", startTime);
-        if (!error.isEmpty()) {
-            errors.add(error);
-        }
+        errors.add(validator.getValidityInfoForNonNullField("submission opening time", startTime));
```

Comment:
Just a query: are these fields defined as constants somewhere? If so, it would be better to use them instead of hardcoded names.